packages/swap/src/routes/v2/utils.ts

```diff
@@ -174,10 +174,15 @@ export const getDepositInfo = (
       txRef: depositTransactionRef,
       txConfirmations: pendingDeposit?.transactionConfirmations,
       witnessedAt:
-        swapRequest?.originType === 'VAULT'
-          ? swapRequest.swapRequestedAt.valueOf()
-          : swapRequest?.depositReceivedAt?.valueOf(),
-      witnessedBlockIndex: swapRequest?.depositReceivedBlockIndex ?? undefined,
+        swapRequest?.depositBoostedAt?.valueOf() ??
+        swapRequest?.depositReceivedAt?.valueOf() ??
+        swapRequest?.swapRequestedAt?.valueOf() ??
```

Comment:
wouldn't this make a swap that's just waiting for deposit to return a wrong value?

Reply:
maybe i missunderstand the question, but if there is no deposit yet there should be no `swapRequest`

Reply:
yes, so we're returning the swapRequest value as witnessed block even though there is no deposit yet, is that correct?

Reply:
a swap request is only created after a deposit is finalized or boosted

Reply:
ah my bad, thanks